src/main/java/seedu/eduke8/storage/TopicsStorage.java

```diff
@@ -15,25 +15,31 @@
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.logging.Level;
 
+import static seedu.eduke8.exception.ExceptionMessages.ERROR_TOPICS_JSON_DESCRIPTION_BLANK;
 import static seedu.eduke8.exception.ExceptionMessages.ERROR_TOPICS_JSON_NOT_FOUR_OPTIONS;
 import static seedu.eduke8.exception.ExceptionMessages.ERROR_TOPICS_JSON_NO_CORRECT;
 import static seedu.eduke8.exception.ExceptionMessages.ERROR_TOPICS_JSON_PREFACE;
 import static seedu.eduke8.exception.ExceptionMessages.ERROR_TOPICS_JSON_QUESTION;
+import static seedu.eduke8.exception.ExceptionMessages.ERROR_TOPICS_JSON_TITLE_BLANK;
+import static seedu.eduke8.exception.ExceptionMessages.ERROR_TOPICS_JSON_TITLE_DUPLICATE;
 import static seedu.eduke8.exception.ExceptionMessages.ERROR_TOPICS_JSON_TOO_MANY_CORRECT;
 import static seedu.eduke8.exception.ExceptionMessages.ERROR_TOPICS_JSON_TOPIC;
 
 public class TopicsStorage extends LocalStorage {
     private boolean wasCorrectAnswerMarked;
     private String currentQuestionDescription;
     private String currentTopicTitle;
+    private final HashSet<String> topicTitles;
```

Comment:
Great implementation to check for duplicated topics! Do you think the same should be done for questions?

Reply:
Implemented this as well

Reply:
thanks!